src/components/Trade/hooks/useSwapper/useSwapper.ts

```diff
@@ -182,13 +191,22 @@ export const useSwapper = () => {
   }, [])
 
   const sellAssetBalance = useAppSelector(state =>
-    selectPortfolioCryptoBalanceByAssetId(state, { assetId: sellTradeAsset?.asset?.assetId ?? '' }),
-  )
-
-  const feeAsset = useAppSelector(state =>
-    selectFeeAssetById(state, sellTradeAsset?.asset?.assetId ?? 'eip155:1/slip44:60'),
+    selectPortfolioCryptoBalanceByAssetId(state, {
+      assetId: sellTradeAsset?.asset?.assetId ?? '',
+    }),
   )
 
+  const osmosisAsset = useAppSelector(state => selectAssetById(state, osmosisAssetId))
+  const feeAsset = useAppSelector(state => {
+    if (!sellTradeAsset?.asset) {
+      return selectFeeAssetById(state, 'eip155:1/slip44:60')
+    } else if (
+      sellTradeAsset.asset.chainId === KnownChainIds.OsmosisMainnet ||
+      sellTradeAsset.asset.chainId === KnownChainIds.CosmosMainnet
+    ) {
+      return selectFeeAssetById(state, osmosisAsset?.assetId)
```

Comment:
If this `else if` section needed? wouldn't `sellTradeAsset.asset.assetId` be the same thing so the else would handle it?

Reply:
feeAsset here is referring to the fee asset we pay to broadcast the transaction.

Its an interesting problem because now we technically have 2 fee assets because we are broadcasting on 2 chains.

Reply:
hmm yea that naming was confusing then.  We should probably rename it, but I believe that's outside the scope of this ticket